hilti/include/rt/exception.h

```diff
@@ -98,6 +103,64 @@ HILTI_EXCEPTION(OutOfRange, RuntimeError)
 /** Exception flagging invalid arguments passed to a function. */
 HILTI_EXCEPTION(InvalidArgument, RuntimeError);
 
+/** Exception flagging access to an iterator that not, or no longer, valid. */
```

Comment:
I feel that these we loose some granularity by putting all these into the same namespace, at least for some of them WDYT about instead keeping them in their origin namespaces and probably headers? For headers currently without impl file we could add a very simple on to emit defintions for the exceptions.

Reply:
I actually like the new naming  structure better because the previous was inconsistent in terms of what got namespaced vs what not, and because I'm not sure that that level of granularity is actually helpful/needed. I was thinking about going even further and replace some of the more specifics ones with more general exceptions; didn't do that for now, but we could still consider that.